test-utils/src/rpc/client.rs

```diff
@@ -0,0 +1,305 @@
+        &self.address
+    }
+
+    /// Access to the inner RPC client for other usage.
+    #[allow(dead_code)]
+    const fn rpc(&self) -> &Rpc<HttpRpc> {
+        &self.rpc
+    }
+
+    /// Request data and map the response to a [`VerifiedBlockInformation`].
+    ///
+    /// # Panics
+    /// This function will panic at any error point, e.g.,
+    /// if the node cannot be connected to, if deserialization fails, etc.
+    pub async fn get_verified_block_information(&self, height: u64) -> VerifiedBlockInformation {
+        #[derive(Debug, Deserialize)]
+        struct Result {
+            blob: String,
+            block_header: BlockHeader,
+        }
+
+        #[derive(Debug, Deserialize)]
+        struct BlockHeader {
+            block_weight: usize,
+            long_term_weight: usize,
+            cumulative_difficulty: u128,
+            hash: String,
+            height: u64,
+            pow_hash: String,
+            reward: u64, // generated_coins + total_tx_fees
+        }
+
+        let result = self
+            .rpc
+            .json_rpc_call::<Result>(
+                "get_block",
+                Some(json!(
+                    {
+                        "height": height,
+                        "fill_pow_hash": true
+                    }
+                )),
+            )
+            .await
+            .unwrap();
+
+        // Make sure this is a trusted, `pow_hash` only works there.
+        assert!(
+        	!result.block_header.pow_hash.is_empty(),
+        	"untrusted node detected, `pow_hash` will not show on these nodes - use a trusted node!"
+        );
+
+        let reward = result.block_header.reward;
+
+        let (block_hash, block) = spawn_blocking(|| {
+            let block = Block::read(&mut hex::decode(result.blob).unwrap().as_slice()).unwrap();
+            (block.hash(), block)
+        })
+        .await
+        .unwrap();
+
+        let txs: Vec<Arc<TransactionVerificationData>> = self
+            .get_transaction_verification_data(&block.txs)
+            .await
+            .map(Arc::new)
+            .collect();
+
+        let block_header = result.block_header;
+        let block_hash_2 = <[u8; 32]>::try_from(hex::decode(&block_header.hash).unwrap()).unwrap();
+        let pow_hash = <[u8; 32]>::try_from(hex::decode(&block_header.pow_hash).unwrap()).unwrap();
+
+        // Assert the block hash matches.
+        assert_eq!(block_hash, block_hash_2);
+
+        let total_tx_fees = txs.iter().map(|tx| tx.fee).sum::<u64>();
+        let generated_coins = block
+            .miner_tx
+            .prefix
+            .outputs
+            .iter()
+            .map(|output| output.amount.expect("miner_tx amount was None"))
+            .sum::<u64>()
+            - total_tx_fees;
+        assert_eq!(
+            reward,
+            generated_coins + total_tx_fees,
+            "generated_coins ({generated_coins}) + total_tx_fees ({total_tx_fees}) != reward ({reward})"
+        );
```

Comment:
Correct calculation of `generated_coins` and `reward` and `total_tx_fees`? (RPC tests all pass)

Reply:
yep, correct